adapters/mgidX/mgidX.go

```diff
@@ -87,7 +95,44 @@ func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.E
 	return adapterRequests, nil
 }
 
+func (a *adapter) getImpressionExt(imp *openrtb2.Imp) (*openrtb_ext.ImpExtMgidX, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, fmt.Errorf("Bidder extension not provided or can't be unmarshalled")
+	}
+
+	var mgidXExt openrtb_ext.ImpExtMgidX
+	if err := json.Unmarshal(bidderExt.Bidder, &mgidXExt); err != nil {
+		return nil, fmt.Errorf("Error while unmarshaling bidder extension")
+	}
+
+	return &mgidXExt, nil
+}
+
+func (a *adapter) buildEndpointURL(params *openrtb_ext.ImpExtMgidX) (string, error) {
+	var endpointParams macros.EndpointTemplateParams
+
+	if params.Region == "eu" {
+		endpointParams = macros.EndpointTemplateParams{Host: "eu"}
+	} else {
+		endpointParams = macros.EndpointTemplateParams{Host: "us-east-x"}
+	}
+
+	return macros.ResolveMacros(a.endpoint, endpointParams)
+}
+
 func (a *adapter) makeRequest(request *openrtb2.BidRequest) (*adapters.RequestData, error) {
+	var mgidXExt *openrtb_ext.ImpExtMgidX
+	mgidXExt, err := a.getImpressionExt(&(request.Imp[0]))
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := a.buildEndpointURL(mgidXExt)
+	if err != nil {
```

Comment:
only `request.Imp[0]` is  considered to prepare `mgidXExt`

@xmgiddev  could you clearify whether adapter supports multi-imps request. If not then should specify this in bidder docs. If multi-imps are supported then should mention that only  `request.Imp[0]` will be considered to prepare `mgidXExt`